### PR TITLE
Launchpad Segment Integration

### PR DESCRIPTION
**Pull request by SSabev**
_Launchpad <> Segment Integration._

We are now adding the ability to send events from segment into [Launchpad.pm](https://www.launchpad.pm) - the mission control for Product teams at scale.

We are adding the following:
* trackEvent
* groupIdentifyUser
* identifyUser

We have added unit tests and have tested this manually.

It requires the following:
* apiRegion - EU by default. We have added the US as means of future extensibility.
* apiSecret - given by Launchpad.pm while onboarding.
* sourceName - to be added.

## Testing

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] confirmed that all events arrive into the Launchpad.pm platform as intended. 
- [ ] [Segmenters] Tested in the staging environment

### CODE REVIEW

**Comment by joe-ayoub-segment** [comment]
hi @SSabev , acknowledging your PR. I'll get to this ASAP.

**Comment by joe-ayoub-segment** [comment]
Hi @SSabev , 
We've recently updated our process for some PR reviews to include a face to face review so that I can provide more relevant feedback. 

The agenda includes getting an understanding of how the customer will use the Integration, what types of data they will send to your platform, and how the customer will use the data once it gets to your platform. 

When I understand this context better we'll discuss the identifiers and other fields you'll be collecting, and we'll walk the code. 

Would you be OK scheduling a call with me via Calendly please? https://calendly.com/joe_ayoub/integration-workshop 

Best regards,
Joe

**Comment by SSabev** [comment]
Hey Joe,

That's now done! Let me know if the time works

**Comment by joe-ayoub-segment** [comment]
Hi @SSabev , 
Thanks for taking the time to meet just now. Here are the notes from our call, including feedback and requested changes: 

Notes: 

- Email is not an identifier. Different profiles can share the same email. 
- Main identifier is called a distinct_id
- A profile can have multiple distinct ids
- Plan is to map anonymousId and userId values as distinct_ids
- groupId is the identifier for an organization (this is the term used by Launchpad) 
- Feedback

Settings

- Secret Key - should be a password type
- Remove Strict Mode 

Fix testAuthentication to return 200 or … or a resolvable promise

Track()

- Explicitly indicate which fields are required or optional 
- Look for $.traits and then $.context.traits
- Need description for the Action 
- Make Timestamp field required 
- Make messageId required. It will always be there. 
- getEventProperties: This line looks incorrect: id: payload.event,
- source mapping looks incorrect: source: integration?.name == 'Iterable' ? 'Iterable' : 'segment',

Identify()

- Description should be updated. 
- identify() calls are often fired when a trait is collected, or when a userId is collected. 
- Maybe use wording: “Creates or updates a user profile, and adds or updates trait values on the user profile…”
- Refactor the perform() function 

group()

- Group Key - Amend description to better explain that group key is a way to connect multiple organizations together 
- groupId field - Should be updated 
- Consider adding anonymousId as a field 
- Handle when there are no traits.

**Comment by joe-ayoub-segment** [comment]
Note: @SSabev updated the PR and sent the following email 5:30pm Friday 3-Feb: 

Hey Joe,

Just FYI:  I pushed all the changes out as requested. Will give it another brush over over the weekend but if you don't hear from me over the weekend assume it's all g2g and ready to test on Monday from your end!

Thanks,
Stefan

**Comment by joe-ayoub-segment** [comment]
Thanks for getting the changes made @SSabev . I'll schedule a review for this to take place ASAP. 
Note: Yesterday was a public holiday where I am based so I am a little behind schedule this week.

**Comment by SSabev** [comment]
@joe-ayoub-segment done, dropped you an email to explain the changes!

File: packages/destination-actions/src/destinations/launchpad/identifyUser/index.ts [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+      type: 'string',
+      description: "The IP address of the user. This is only used for geolocation and won't be stored.",
+      default: {
+        '@path': '$.context.ip'
+      }
+    },
+    userId: {
+      label: 'User ID',
+      type: 'string',
+      required: true,
```
Comment (joe-ayoub-segment) [inline packages/destination-actions/src/destinations/launchpad/identifyUser/index.ts:24]:
```suggestion
```

Comment (joe-ayoub-segment) [reply packages/destination-actions/src/destinations/launchpad/identifyUser/index.ts:24]: Hi @SSabev the userId should not be required, should it?

Comment (joe-ayoub-segment) [reply packages/destination-actions/src/destinations/launchpad/identifyUser/index.ts:24]: Even for an identify() call the userId is not required. Users can pass in traits via an identify() call and they don't need to include a userId.

**Comment by SSabev** [comment]
@joe-ayoub-segment fixed and g2g

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @SSabev , 
The deployment didn't go out today due to an issue unrelated to your PR. It should be deployed later today.

**Comment by joe-ayoub-segment** [comment]
Hi @SSabev , 
Unfortunately the deployment yesterday was cancelled (nothing related to your PR). The team plans to release next week. I'll confirm the date when I know more. Apologies for the inconvenience.

**Comment by joe-ayoub-segment** [comment]
Hi @SSabev , 

This PR was deployed yesterday. You should be able to start testing it now!
It's available in Private Beta, so you'll need to access it via a specific URL.
https://app.segment.com/YOUR_WORKSPACE_SLUG/destinations/catalog/actions-launchpad

If you've any follow up questions, including regarding adding meta data and documentation please email [partner-support@segment.com](mailto:partner-support@segment.com).

**Pull request merged**